RoslynPluginGenerator/AnalyzerPluginGenerator.cs

```diff
@@ -98,6 +104,33 @@ public bool Generate(NuGetReference analyzeRef, string language, string sqaleFil
         }
 
         /// <summary>
+        /// Recursively checks a package and all dependencies for the presence of the RequireLicenseAcceptance flag.
+        /// Returns true if any of the packages in the dependency tree require license acceptance.
+        /// </summary>
+        /// <param name="package">The package to test.</param>
+        /// <returns></returns>
+        private bool PackageRequiresLicenseAcceptance(IPackage package)
+        {
+            if (package.RequireLicenseAcceptance)
+            {
+                return true;
+            }
+            foreach (PackageDependencySet dependencySet in package.DependencySets)
+            {
+                foreach (PackageDependency dependency in dependencySet.Dependencies)
+                {
+                    // Also check that no dependencies require license acceptance
+                    IPackage dependencyPkg = packageHandler.FetchPackage(dependency.Id, dependency.VersionSpec.MaxVersion);
```

Comment:
Well, in this case only one version of the dependency is ever available (the CreatePackage helper method is hard-coded to "1.0"). But this could be different in a real use case.

When the dependency was created only one version ("1.0") was specified, but NuGet clearly does some internal processing.